flocker/control/httpapi.py

```diff
@@ -492,11 +498,15 @@ def create_container_configuration(self, host, name, image, ports=()):
                 external_port=port['external']
             ))
 
+        if environment:
```

Comment:
:question: Surely if `environment` is not `None`, it's guaranteed to be a dictionary because the JSON schema type checking has already been enforced before we get to the endpoint?